app-website-4/ProductsApi/Auth/TokenProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.IdentityModel.Tokens;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+
+namespace ProductsApi.Auth
+{
+    public class TokenProvider
+    {
+        public string GenerateToken(string[] usernameAndPassword)
+        {
+            if (usernameAndPassword[0] == "admin" && usernameAndPassword[1] == "password")
+            {
+                var claimsdata = new[] { new Claim(ClaimTypes.Name, usernameAndPassword[0]) };
+                var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes("This is a secret key"));
+                var signInCred = new SigningCredentials(key, SecurityAlgorithms.HmacSha256Signature);
+                var token = new JwtSecurityToken(
+                     issuer: "Products API",
+                     audience: "Products App",
```

Comment:
I'd extract these as constants somewhere tbh.

Also in this case, your issuer and audience are the same.

The Products API issues token after successful "authentication" of a username/password combo. It can also consume the tokens it's issued to authorize clients to use its API.

Reply:
Ah, OK, that helps me understand this better, thanks.